compound/Chemical.java

```diff
@@ -0,0 +1,149 @@
+package compound;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import java.util.Stack;
+
+
+
+public class Chemical {
```

Comment:
1. We can create a parser to parse the given molecular formula in a different class, add proper exception handling for wrong input. eg: brackets mismatch.
2. We can create an enum to store the tokens used in the molecular formula